src/shared/Core/CommandContext.cs

```diff
@@ -92,7 +92,7 @@ public CommandContext(string appPath, string installDir)
 
             Streams = new StandardStreams();
             Trace   = new Trace();
-            Trace2  = new Trace2();
+            Trace2  = new Trace2(this);
```

Comment:
At first view I raised my eyebrow about passing the entire `ICommandContext` to any component that hangs off of `CommandContext` itself, but then noticed that we actually do have one component that does this: `CredentialStore`.

I think we should avoid this pattern unless we're creating `Trace2` as the last thing in the `CommandContext` constructor.

Imagine if, inside of `Trace2.ctor` we try and access a service or component on `ICommandContext` that hasn't been setup yet?

I think `Trace2` only needs `IEnvironment` right now.